### PR TITLE
tsduck 3.37-3670

### PR DESCRIPTION
**Pull request by Porkepix**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by lelegard** [comment]
Hi @Porkepix 

I was about to create that PR but you preceded me. There is one problem in your PR: a missing dependency. Because of new features in this version of tsduck, there new requirements. You need to add this line in the formula:
~~~
  depends_on "openssl@3"
~~~

**Comment by lelegard** [comment]
> I sure can add it, however shouldn't a missing dependency like that one trigger some sort of error on either build or tests? (Two jobs already passed the tests)

OpenSSL is certainly already installed on all runners and build won't fail during the initial validation phase. However, this is a new explicit dependency for this version of tsduck. It is required. So, please add it.

Also, before sending a PR on a project, please check if the formula must be be updated somehow, as it is the case in this version.

**Comment by chenrui333** [comment]
The openssl@3 is actually a transitive dependency as you can see in the formula dep tree

```
$ brew deps --tree tsduck
tsduck
├── librist
│   ├── cjson
│   ├── libmicrohttpd
│   │   └── gnutls
│   │       ├── ca-certificates
│   │       ├── gmp
│   │       ├── libidn2
│   │       │   ├── libunistring
│   │       │   └── gettext
│   │       ├── libtasn1
│   │       ├── libunistring
│   │       ├── nettle
│   │       │   └── gmp
│   │       ├── p11-kit
│   │       │   ├── ca-certificates
│   │       │   └── libtasn1
│   │       └── unbound
│   │           ├── libevent
│   │           │   └── openssl@3
│   │           │       └── ca-certificates
│   │           ├── libnghttp2
│   │           └── openssl@3
│   │               └── ca-certificates
│   └── mbedtls
├── libvatek
│   └── libusb
└── srt
    └── openssl@3
        └── ca-certificates
```

**Comment by lelegard** [comment]
Look at this, you see that the dependency is required. All other dependencies are already listed in the formula.
~~~
$ otool -L libtsduck.dylib
libtsduck.dylib:
        @rpath/libtsduck.dylib (compatibility version 0.0.0, current version 0.0.0)
        /usr/local/opt/libvatek/lib/libvatek_core.dylib (compatibility version 0.0.0, current version 0.0.0)
        /System/Library/Frameworks/PCSC.framework/Versions/A/PCSC (compatibility version 0.0.0, current version 602.60.5)
==>     /usr/local/opt/openssl@3/lib/libcrypto.3.dylib (compatibility version 3.0.0, current version 3.0.0)
        /usr/lib/libedit.3.dylib (compatibility version 2.0.0, current version 3.0.0)
        /usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 9.0.0)
        /usr/local/opt/srt/lib/libsrt.1.5.dylib (compatibility version 1.5.0, current version 1.5.3)
        /usr/local/opt/librist/lib/librist.4.dylib (compatibility version 4.0.0, current version 4.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.0.0)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.36.0)

~~~

**Comment by lelegard** [comment]
> The openssl@3 is actually a transitive dependency as you can see in the formula dep tree

@chenrui333, this only explains why the tests didn't fail. But transitive dependencies may change or disappear. Starting with this version, tsduck explicitly requires openssl. If some new version of tsduck decides to drop support for rist and srt, the transitive dependency to openssl will disappear but other features of tsduck will still requires openssl.

The explicit dependency must be added.

**Comment by lelegard** [comment]
@chenrui333, it is funny that `gnutls` requires `openssl` (even transitively, again) because the former is supposed to be a replacement for the latter 😄 

Even more "fun": `librist` explicitly depends on `mbedtls` (yet another TLS implementation) and on `libmicrohttpd` which depends on `gnutls`, which (transitively) depends on `openssl`. Therefore, using `librist` brings the three main SSL/TLS libraries altogether in the same memory space.

**Comment by chenrui333** [comment]
> Starting with this version, tsduck explicitly requires openssl. 

Make sense, 👍 

> Therefore, using `librist` brings the three main SSL/TLS libraries altogether in the same memory space.

:(

**Comment by lelegard** [comment]
@chenrui333, could you please direct me to some doc or explanation about autobump?
I cannot find any and Google floods me with irrelevant stuff.

**Comment by chenrui333** [comment]
> @chenrui333, could you please direct me to some doc or explanation about autobump?

it is the autobump process managed by homebrew-core (but if you guys setup the autobump process, we can remove from the one from the core), it is not well documented yet, but you can find the github action workflow, which does that.

**Comment by lelegard** [comment]
> it is the autobump process managed by homebrew-core (but if you guys setup the autobump process, we can remove from the one from the core), it is not well documented yet, but you can find the github action workflow, which does that.

The workflow is not easy to understand. My understanding is that any formula in `autobump.txt` will be automatically updated when a new release of the corresponding project is published. Is that correct?

If this is correct, this is absolutely fine with me since most versions of `tsduck` don't need any modification of the formula, other that url and sha. I just have a question about specific rare cases (like this current new version) where the formula needs an update, such as a new `depends_on`. How does it work?

**Comment by lelegard** [comment]
> What's in autobump.txt will be run through every 3 hours by the bot to check last version available with livecheck and if a new one is available, it'll bump, assuming there's no issue with the check.

Sorry for asking stupid questions, but I am not familiar with this. What does "last version available with livecheck" mean? Which "last version" are we referring to? How does a project advertise a "new version", outside creating a release on GitHub?

> to have it automatically when a new release is published, you have to include that in your workflows and have the action run at release.

Do you have an example of GitHub project which does this?

File: .github/autobump.txt [hunk 1]
```diff
@@ -1723,6 +1723,7 @@ trino
 trivy
 trojan-go
 trufflehog
+tsduck
```
Comment (carlocab) [inline .github/autobump.txt:1726]: Didn't follow the discussion about adding this to the autobump list, so feel free to open a new PR to remove it if it's not suitable.

**Review by carlocab — APPROVED** [review]

**Comment by lelegard** [comment]
> for well-known cases such as GitHub here I'm not sure if it relies on tags on releases

It seems to work. Considering that this PR is not yet merged, tsduck is not yet available on HomeBrew but the tag and release are present in GitHub:
~~~
$ brew livecheck tsduck
tsduck: 3.36-3528 ==> 3.37-3670
~~~

File: .github/autobump.txt (hunk 1 above)
Comment (lelegard) [reply .github/autobump.txt:1726]: Thanks @carlocab, it is fine with me.

I am not familiar with autobump. My understanding is that the brew bot will automatically detect a new "version" (tag? release?) of the upstream project on GitHub and will autobump in HomeBrew without need to submit a PR. Is this correct?

**Pull request merged**